src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -193,64 +214,185 @@ We support
+                    $"Invalid value '{rawEnabled}' for field '{AzureAppConfigurationFeatureFlagFields.Enabled}' of feature '{featureName}'.");
+            }
 
-            if (featureManagementConfigurationSection.Exists())
+            if (enabled)
             {
-                return featureManagementConfigurationSection.GetChildren();
+                IEnumerable<IConfigurationSection> filterSections = conditions.GetSection(AzureAppConfigurationFeatureFlagFields.ClientFilters).GetChildren();
+
+                if (filterSections.Any())
+                {
+                    foreach (IConfigurationSection section in filterSections)
+                    {
+                        //
+                        // Arrays in json such as "myKey": [ "some", "values" ]
+                        // Are accessed through the configuration system by using the array index as the property name, e.g. "myKey": { "0": "some", "1": "values" }
+                        if (int.TryParse(section.Key, out int _) && !string.IsNullOrEmpty(section[AzureAppConfigurationFeatureFlagFields.Name]))
+                        {
+                            enabledFor.Add(new FeatureFilterConfiguration()
+                            {
+                                Name = section[AzureAppConfigurationFeatureFlagFields.Name],
+                                Parameters = new ConfigurationWrapper(section.GetSection(AzureAppConfigurationFeatureFlagFields.Parameters))
+                            });
+                        }
+                    }
+                }
+                else
+                {
+                    enabledFor.Add(new FeatureFilterConfiguration
+                    {
+                        Name = "AlwaysOn"
+                    });
+                }
             }
 
+            return new FeatureDefinition()
+            {
+                Name = featureName,
+                EnabledFor = enabledFor,
+                RequirementType = requirementType
+            };
+        }
+
+        private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
+        {
+            IConfigurationSection featureManagementConfigurationSection = _configuration.GetSection(ConfigurationFields.FeatureManagementSectionName);
+
+            if (!featureManagementConfigurationSection.Exists())
+            {
+                if (!RootConfigurationFallbackEnabled)
+                {
+                    Logger?.LogDebug($"No configuration section named '{ConfigurationFields.FeatureManagementSectionName}' was found.");
+
+                    return Enumerable.Empty<IConfigurationSection>();
+                }
+
+                featureManagementConfigurationSection = _configuration as IConfigurationSection;
+            }
+
+            IConfigurationSection featureFlagsConfigurationSection = featureManagementConfigurationSection.GetSection(AzureAppConfigurationFeatureFlagFields.FeatureFlagsSectionName);
```

Comment:
`as` implies null as a possibility. Meaning this is a possible null-ref.

Reply:
Are you expecting a null check after the type casting?
The _configuration is IConfiguration. The null check for _configuration happens in the constructor of ConfigurationFeatureDefinitionProvider. So _configuration cannot be null here and it can always be casted to IConfigurationSection.

Reply:
`as` is not a type cast.

this is a type cast

`(IConfigurationSection)_configuration`

Yes, if `as` is used, then a null check is needed. If we know that the object IS of a certain type then we shouldn't use `as` and we should type cast instead.

Reply:
I think we can use type cast `IConfigurationSection featureFlagsConfigurationSection = (IConfigurationSection)_configuration` directly.

Reply:
Okay, type case it is.